lib/metadataTypes/Script.js

```diff
@@ -141,32 +140,92 @@ class Script extends MetadataType {
-     * @param {TYPE.TemplateMap} variables variables to be replaced in the metadata
+     * @param {TYPE.TemplateMap} templateVariables variables to be replaced in the metadata
      * @param {string} templateName name of the template to be built
-     * @returns {Promise} Promise
+     * @returns {Promise.<void>} Promise
      */
-    static async buildDefinitionForExtracts(
+    static buildDefinitionForExtracts(
         templateDir,
         targetDir,
         metadata,
-        variables,
+        templateVariables,
         templateName
+    ) {
+        return this._buildXForExtracts(
+            templateDir,
+            targetDir,
+            metadata,
+            templateVariables,
+            templateName,
+            'definition'
+        );
+    }
+    /**
+     * helper for buildTemplate
+     * handles extracted code if any are found for complex types
+     *
+     * @param {string} templateDir Directory where metadata templates are stored
+     * @param {string|string[]} targetDir (List of) Directory where built definitions will be saved
+     * @param {TYPE.ScriptItem} metadata main JSON file that was read from file system
+     * @param {TYPE.TemplateMap} templateVariables variables to be replaced in the metadata
+     * @param {string} templateName name of the template to be built
+     * @returns {Promise.<void>} void
+     */
+    static buildTemplateForExtracts(
+        templateDir,
+        targetDir,
+        metadata,
+        templateVariables,
+        templateName
+    ) {
+        return this._buildXForExtracts(
+            templateDir,
+            targetDir,
+            metadata,
+            templateVariables,
+            templateName,
+            'template'
+        );
+    }
+
+    /**
+     * helper for buildTemplateForExtracts / buildDefinitionForExtracts
+     * handles extracted code if any are found for complex types
+     *
+     * @param {string} templateDir Directory where metadata templates are stored
+     * @param {string|string[]} targetDir (List of) Directory where built definitions will be saved
+     * @param {TYPE.ScriptItem} metadata main JSON file that was read from file system
+     * @param {TYPE.TemplateMap} templateVariables variables to be replaced in the metadata
+     * @param {string} templateName name of the template to be built
+     * @param {'definition'|'template'} mode defines what we use this helper for
+     * @returns {Promise.<void>} Promise
+     */
+    static async _buildXForExtracts(
+        templateDir,
+        targetDir,
+        metadata,
+        templateVariables,
+        templateName,
+        mode
     ) {
         // get SSJS from filesystem
         let code = await this._mergeCode(metadata, templateDir, templateName);
-        // replace template variables with their values
         try {
-            code = Mustache.render(code, variables);
+            if (mode === 'definition') {
```

Comment:
I would probably make this a ternary operator given we only have two scenarios? otherwise, i would expect an else block with an error

Reply:
I find it easier to understand the logic given that this isbthe one and only place where we do sth different for buildTemplate vs buildDefinition. The fact that it is an internal helper and even jsdoc defined proper values leads me to think that an else-block is wasting lines here?